src/main/java/com/sfuentes/Student.java

```diff
@@ -0,0 +1,30 @@
+package com.sfuentes;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Student {
+  String[][] students;
+  Map<String, Integer> categories = new HashMap<>();
+  int menCount;
+  int womanCount;
```

Comment:
These three variables are only used in the sortByWomenAndMen hence they should be local variables in that method instead of being properties of the class.

Reply:
fixed